### ISSUE TITLE
NLog network targets (udp) fail after NlogInspector attaches log target

### ISSUE THREAD

**Issue by thirkcircus** [rho24/Glimpse.NLog#16]
I have a web app which logs to the NLogViewer target (essentially fires off udp to a listener).

When Glimpse.Nlog is included in the web app, the process of attaching the Glimpse target to the NLog config is causing NLog NetworkTargets to be closed - and subsequent calls to them fail with a first time exception swallowed by NLog.

![image](https://cloud.githubusercontent.com/assets/81165/7384790/ea3b7d18-ee8d-11e4-969b-15d4915a10a0.png)

It appears the root cause is that calling 
```csharp
LogManager.Configuration = LogManager.Configuration; //Re initialize configuration.
```
as per the changes made to fix https://github.com/rho24/Glimpse.NLog/issues/5,
causes NLog to close existing network connections - and these turn out not to be reopened.  When a network target is configured to reuse existing connections, this causes a NullRef exception every time trying to log to the network target - and it also causes other targets in the chain to fail.

**Comment by thirkcircus** [comment]
forgot to mention, what I'm seeing is a pile of First time exceptions thrown (and swallowed) by NLog, and no logs written to any target - and the nlog debug output shows this;

2015-04-29 11:52:33.8554 Error Exception in asynchronous handler NLog.NLogRuntimeException: Exception occurred in NLog ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at NLog.Internal.NetworkSenders.UdpNetworkSender.DoSend(Byte[] bytes, Int32 offset, Int32 length, AsyncContinuation asyncContinuation)
   at NLog.Targets.NetworkTarget.<>c__DisplayClass13.<ChunkedSend>b__12(Exception ex)
   at NLog.Targets.NetworkTarget.Write(AsyncLogEventInfo logEvent)
   at NLog.Targets.Target.WriteAsyncLogEvent(AsyncLogEventInfo logEvent)
   --- End of inner exception stack trace ---
   at NLog.LoggerImpl.<>c__DisplayClass1.<Write>b__0(Exception ex)
   at NLog.Internal.SingleCallContinuation.Function(Exception exception)

debugging into nlog itself showed the UdpNetworkSender private socket was null - and this was because it had been closed as shown in the call stack above.

**Comment by rho24** [comment]
Hi Justin
The pull request looks fine, but leaves me wondering why I moved away from the Reload() method in the first place. I'll give it a thorough test and push a new release asap, might be a few days though as I currently don't have internet at home.
Cheers for the contribution.
Rich

**Comment by thirkcircus** [comment]
Yeah, I don't know the nlog api that well but it seems there might be a difference between
```csharp
LogManager.Configuration.Reload(); // you had previously
// and
LogManager.ReconfigExistingLoggers(); // I have changed to this
```

the former causes the old config to be unloaded (and network senders closed) whereas the latter just resets any existing loggers with the additional targets/filters defined by Glimpse.

**Comment by rho24** [comment]
Hi Justin
Sorry had to delay release as there was an issue with MyGet. The fix is now on nuget in version 1.6.1.
Thanks again
